src/compiler/scanner.ts

```diff
@@ -997,6 +997,19 @@ export function isIdentifierText(name: string, languageVersion: ScriptTarget | u
     return true;
 }
 
+const enum EscapeSequenceScanningFlags {

```

Comment:
Actually, there are only 8 valid states:
```
String,
String | ReportErrors,
RegularExpression,
RegularExpression | AnnexB,
RegularExpression | AnyUnicodeMode,
RegularExpression | AtomEscape,
RegularExpression | AnnexB | AtomEscape,
RegularExpression | AnyUnicodeMode | AtomEscape,
```
which is supposed to fit in 3 booleans.

Reply:
I think this works fine.